redis/src/main/java/org/apache/zeppelin/redis/RedisInterpreter.java

```diff
@@ -0,0 +1,1986 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.redis;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.apache.zeppelin.interpreter.Interpreter;
+import org.apache.zeppelin.interpreter.InterpreterContext;
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import redis.clients.jedis.BinaryClient;
+import redis.clients.jedis.BitOP;
+import redis.clients.jedis.GeoUnit;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.apache.zeppelin.interpreter.InterpreterResult.Code.ERROR;
+import static org.apache.zeppelin.interpreter.InterpreterResult.Code.SUCCESS;
+
+/**
+ * Redis interpreter for Zeppelin.
+ */
+public class RedisInterpreter extends Interpreter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RedisInterpreter.class);
+
+  private static final char SPACE = ' ';
+  private static final char DOUBLE_QUOTATION = '\"';
+
+  private static final String REDIS_HOST = "redis.host";
+  private static final String REDIS_PORT = "redis.port";
+
+  private String host = Protocol.DEFAULT_HOST;
+  private int port = Protocol.DEFAULT_PORT;
+
+  private JedisPoolConfig config;
+  private JedisPool clientPool;
+
+  public RedisInterpreter(Properties property) {
+
+    super(property);
+
+    if (property.containsKey(REDIS_HOST) &&
+            !Strings.isNullOrEmpty(property.getProperty(REDIS_HOST))) {
+      host = property.getProperty(REDIS_HOST);
+    }
+
+    if (property.containsKey(REDIS_PORT) &&
+            !Strings.isNullOrEmpty(property.getProperty(REDIS_PORT))) {
+      port = Integer.valueOf(property.getProperty(REDIS_PORT));
+    }
+
+    LOG.info("Connection to %s:%d", host, port);
+
+    // TODO(darion) : Need more config ?
+    config = new JedisPoolConfig();
+    config.setMaxTotal(JedisPoolConfig.DEFAULT_MAX_TOTAL);
+    config.setMaxIdle(JedisPoolConfig.DEFAULT_MAX_IDLE);
+    config.setMinIdle(JedisPoolConfig.DEFAULT_MIN_IDLE);
+    config.setMaxWaitMillis(JedisPoolConfig.DEFAULT_MAX_WAIT_MILLIS);
+  }
+
+  @Override
+  public void open() {
+    clientPool = new JedisPool(config, host, port);
+  }
+
+  @Override
+  public InterpreterResult interpret(String input, InterpreterContext context) {
+    String[] tokens = splitTokens(input);
+    String keyword = tokens[0];
+
+    InterpreterResult result;
+    switch (Protocol.Command.valueOf(keyword)) {
```

Comment:
@anthonycorbacho  @HeartSaVioR  It's seems a good idea to split execute Redis Command into classes and this command classes will reused in  Redis Cluster Interpreter. I have create a abstract class `RedisCommand` as a basic class : 

```
public abstract class RedisCommand {

    private JedisPool clientPool;

    public RedisCommand(JedisPool clientPool) {
        this.clientPool = clientPool;
    }

    private Jedis borrowClient() {
        return clientPool.getResource();
    }

    private void releaseClient(Jedis client) {
        client.close();
    }

    public InterpreterResult execute(String[] tokens) {
        Jedis client = borrowClient();
        try {
            return execute(client, tokens);
        } finally {
            releaseClient(client);
        }
    }

    /**
     * @param tokens
     * @return
     * @throws Exception
     */
    protected abstract InterpreterResult execute(Jedis client, String[] tokens);

}
```

I think it's OK and each redis command could extend this one .